AmericanMedicalAssociation/ama-style-guide-2
# Bugfix/ewl 7856 unordered lists only 2 levels

## Ticket(s)
[Unordered lists support more than two levels of indent](https://issues.ama-assn.org/browse/EWL-7856)

**Github Issue**
N/A

**Jira Ticket**
- [Unordered lists support more than two levels of indent](https://issues.ama-assn.org/browse/EWL-7856)

## Description
refactored list mixins and added needed classes to restrict application.


## To Test
- [x] set up d8 for local styleguide development and pull complimentary branch: https://github.com/AmericanMedicalAssociation/ama-d8/pull/1866
- [x] pull this branch and run `gulp serve`
- [x] Follow these instructions, and for each block, try to add 1-2-3 nested lists. In the wysiwyg, it should show up to 1 nested level but no more. After you save, custom blocks should show only one top level list.

- Go to http://ama-one.local/block/2091?destination=/admin/structure/block/block-content%3Finfo%3Dinsurance%26type%3DAll and add a list, then go to http://ama-one.local/delivering-care and make sure they display in the AMA insurance box on the right rail
- Go to http://ama-one.local/block/2076?destination=/admin/structure/block/block-content%3Finfo%3Dsubscribe%26type%3DAll and add a list. check that MR subscribe blocks contain a list (The footer will not contain a list, that block is hardcoded."
- Go to http://ama-one.local/block/2506?destination=/admin/structure/block/block-content%3Finfo%3Dhome%26type%3DAll and add a list to one of the text fields. Go to the bottom of the homepage and check the products block to ensure the list displays
- Go to http://ama-one.local/block/2931?destination=/admin/structure/block/block-content%3Finfo%3D404%2520error%2520page%2520promos%26type%3DAll and add a list to one of the fields. Go to the 404 page to ensure the lists display: http://ama-one.local/paragraph/55241

In addition to the above testing:
- [x] add 1-2-3 levels of nested lists to each node type Evergreen, press release, news article, people bio, hub page.
- [x] ensure that only 1 level of nesting occurs.
- [x] ensure that on news articles, when you scroll down, then back up, the share icons remain against the right page margin.
- [x] ensure that when you scroll down and the share icons switch to the left side, they are not overlapping the text.
- [x] ensure that on event detail and resource pages, that the tab lists look correct and are on the left page margin
- [x] ensure that lists inside the resource and event tabs display correctly.
- [x] ensure that on the homepage, all list elements are against the left margin (ask if you have questions about this.)

## Visual Regressions
N/A


## Relevant Screenshots/GIFs
N/A


## Remaining Tasks
N/A


## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
